javascript/ql/lib/semmle/javascript/dataflow/Configuration.qll

```diff
@@ -2002,6 +2004,42 @@ private class AdditionalBarrierGuardCall extends AdditionalBarrierGuardNode, Dat
   override predicate appliesTo(Configuration cfg) { f.appliesTo(cfg) }
 }
 
+/**
+ * A sanitizer where an inner sanitizer is compared against a boolean.
+ * E.g. (assuming `sanitizes(e)` is an existing sanitizer):
+ * ```javascript
+ * if (sanitizes(e) === true) {
+ *  // e is sanitized
+ * }
+ * ```
+ */
+private class CallAgainstEqualityCheck extends AdditionalBarrierGuardNode {
+  DataFlow::BarrierGuardNode prev;
+  boolean polarity;
+
+  CallAgainstEqualityCheck() {
+    prev instanceof DataFlow::CallNode and
+    exists(EqualityTest test, BooleanLiteral bool |
+      this.asExpr() = test and
+      test.hasOperands(prev.asExpr(), bool) and
+      polarity = test.getPolarity().booleanXor(bool.getBoolValue())
```

Comment:
I'm not sure I understand the use of `xor` here.

Reply:
I need `polarity` to flip both if `test` flips and if `bool` flips.  
`xor` always flips when one of it's inputs flip, that's why I'm using it.  

Then I had to figure out whether I had to add a `.booleanNot()` to the output or not, and that was just done by testing.